agent/otel/otlpmqttexporter/otlp.go

```diff
@@ -0,0 +1,118 @@
+package otlpmqttexporter
+
+import (
+	"context"
+	"crypto/tls"
+	"errors"
+	"fmt"
+	mqtt "github.com/eclipse/paho.mqtt.golang"
+	"go.opentelemetry.io/collector/consumer/consumererror"
+	"net/http"
+	"net/url"
+	"runtime"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/pmetric/pmetricotlp"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.uber.org/zap"
+)
+
+type exporter struct {
+	// Input configuration.
+	config     *Config
+	client     *http.Client
+	tracesURL  string
+	metricsURL string
+	logsURL    string
+	logger     *zap.Logger
+	settings   component.TelemetrySettings
+	// Default user-agent header.
+	userAgent string
+}
+
+// Crete new exporter.
+func newExporter(cfg config.Exporter, set component.ExporterCreateSettings) (*exporter, error) {
+	oCfg := cfg.(*Config)
+
+	if oCfg.Address != "" {
+		_, err := url.Parse(oCfg.Address)
+		if err != nil {
+			return nil, errors.New("address must be a valid mqtt server")
+		}
+	}
+
+	userAgent := fmt.Sprintf("%s/%s (%s/%s)",
+		set.BuildInfo.Description, set.BuildInfo.Version, runtime.GOOS, runtime.GOARCH)
+
+	// Client construction is deferred to start
+	return &exporter{
+		config:    oCfg,
+		logger:    set.Logger,
+		userAgent: userAgent,
+		settings:  set.TelemetrySettings,
+	}, nil
+}
+
+// start actually creates the HTTP client. The client construction is deferred till this point as this
```

Comment:
minor, fix comment. substitute MQTT for HTTP and describe that client is only created if not passed in?

Reply:
will do!